app/middleware/setLocals.js

```diff
@@ -0,0 +1,14 @@
+function fromRequest(req, res, next) {
+  const location = req.query.location;
```

Comment:
I was going to do that but didn't as I thought it was more clear this way. Obviously not.